src/test/java/org/jabref/logic/importer/util/GrobidServiceTest.java

```diff
@@ -0,0 +1,45 @@
+package org.jabref.logic.importer.util;
+
+import org.jabref.preferences.JabRefPreferences;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class GrobidServiceTest {
+
+    static GrobidService grobidService;
+
+    @BeforeAll
+    public static void setup() {
+        grobidService = new GrobidService(JabRefPreferences.getInstance());
+    }
+
+    @Test
+    public void processValidCitationTest() throws GrobidServiceException {
+        String response = grobidService.processCitation("Derwing, T. M., Rossiter, M. J., & Munro, " +
+                "M. J. (2002). Teaching native speakers to listen to foreign-accented speech. " +
+                "Journal of Multilingual and Multicultural Development, 23(4), 245-259.", 1);
+        assertNotNull(response);
+        assertTrue(response.charAt(0) == '@');
+    }
+
+    @Test
+    public void processEmptyStringTest() throws GrobidServiceException {
+        String response = grobidService.processCitation(" ", 1);
+        assertNotNull(response);
+        assertTrue(response.equals(""));
+    }
+
+    @Test
+    public void processInvalidCitationTest() throws GrobidServiceException {
+        String response = grobidService.processCitation("iiiiiiiiiiiiiiiiiiiiiiii", 1);
+        assertNotNull(response);
+        assertTrue(response.equals("@misc{-1,\n" +
+                "\n" +
+                "}\n"));
+    }
+
```

Comment:
It's nice that you have separate tests for the fetcher and the grobid service. It would be nice if you could also add an integration test (new class) that verifies that both modules are interacting together as expected.

Reply:
Thank you a lot for that advice but what exactly do you mean with an integration test in this case? The GrobidService class only has one method which is tested in the GrobidServiceTest class.
There it is proven that the correct value of the GrobidService class is returned.
Or am i overlooking something? :-)

Reply:
An integration test verifies that the modules/classes are interacting correctly. In the present case, this is a test where you input a citation string and verify that the correct bibentry is returned (without mocking the service, so that the Grobid server is used).

Reply:
This will probably require the Docker image integrated in the GitHub workflow. See the test and test-oracle for an example. I assume that it should not be that hard. Think, the MySQL example in the test yml is the best one since it launches a separate DB using an "external" Docker image (doesn't it?)

Reply:
I thought it just queries our test server http://grobid.cm.in.tum.de:8070/ similar to all the other fetcher tests. Traffic overhead should be minimal and it is the easiest solution.

Reply:
ok, so just something like GrobidCitationFetcherTest but without mocking the service. As I understand this new test should be also tagged as FetcherTest, since it relies on a working internet connection.

Reply:
Yes 👍